### PR TITLE
[FIX] discuss: show connection state even without rtcSession

### PR DESCRIPTION
**Pull request by ThanhDodeurOdoo**
Before this commit, a rtcSession was required to display the connection state. Which meant that the receiving end of a peer-to-peer offer was not displaying the connection state until the offer was received, which was hiding early connection issues (pre-offer) like missing bus messages.

### CODE REVIEW

**Review by seb-odoo — COMMENTED** [review]
Ok for the card of other users, but it seems for the current user it always displays the warning when it shouldn't (at least not when being alone or in p2p, I'm not sure about sfu).

File: addons/mail/static/src/discuss/call/common/call_participant_card.js [hunk 1]
```diff
@@ -78,10 +79,21 @@ export class CallParticipantCard extends Component {
...
-                !HIDDEN_CONNECTION_STATES.has(this.rtcSession.connectionState)
+        if (!this.rtcSession) {
+            return false;
+        }
+        if (!this.isOfActiveCall) {
+            return false;
+        }
+        if (HIDDEN_CONNECTION_STATES.has(this.rtcSession.connectionState)) {
+            return false;
+        }
```
Comment (alexkuhn) [inline addons/mail/static/src/discuss/call/common/call_participant_card.js:90]: Nitpicking: could disjoin the conditionals with `||` to have less LOCs for a single `return false;`.

File: addons/mail/static/src/discuss/call/common/call_participant_card.js [hunk 2]
```diff
@@ -78,10 +79,19 @@ export class CallParticipantCard extends Component {
...
+            HIDDEN_CONNECTION_STATES.has(this.rtcSession.connectionState)
+        ) {
+            return false;
+        }
+        if (this.rtcSession.peerConnection) {
+            return true;
+        }
+        return (
+            this.rtcSession.eq(this.rtc?.selfSession) !==
+            (this.rtc.connectionType === CONNECTION_TYPES.P2P)
```
Comment (alexkuhn) [inline addons/mail/static/src/discuss/call/common/call_participant_card.js:94]: This looks equivalent and simpler to understand:
```js
if (this.rtc.connectionType === CONNECTION_TYPES.SERVER) {
   return true;
}
return this.rtcSession.notEq(this.rtc?.selfSession)
```

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Comment by alexkuhn** [comment]
@robodoo r+ delegate+

**Pull request closed without merging**